src/backend/import/importTransactions.ts

```diff
@@ -19,27 +19,25 @@ type ScrapingConfig = Config['scraping'];
 const TRANSACTION_STATUS_COMPLETED = 'completed';
 
 export async function scrapeFinancialAccountsAndFetchTransactions(scrapingConfig: ScrapingConfig, startDate: Date, eventPublisher: EventPublisher) {
-  const companyIdToTransactions: Record<string, EnrichedTransaction[]> = {};
-
   const dowloadedChrome = await getChrome(userDataPath, (percent) => emitChromeDownload(eventPublisher, percent));
 
-  const accountsToScrape = scrapingConfig.accountsToScrape.filter((accountToScrape) => accountToScrape.active !== false);
-  const scrapingPromises = accountsToScrape.map(async (accountToScrape) => {
-    const companyId = accountToScrape.key;
-    try {
-      await eventPublisher.emit(EventNames.IMPORTER_START, buildImporterEvent(accountToScrape, { message: 'Importer start' }));
-      const scrapeResult = await fetchTransactions(accountToScrape, startDate, scrapingConfig, eventPublisher, dowloadedChrome);
-      const transactions = await postProcessTransactions(accountToScrape, scrapeResult);
-      companyIdToTransactions[companyId] = transactions;
-      await eventPublisher.emit(EventNames.IMPORTER_END, buildImporterEvent(accountToScrape, { message: 'Importer end', status: AccountStatus.DONE }));
-    } catch (error) {
-      await eventPublisher.emit(EventNames.IMPORTER_ERROR, buildImporterEvent(accountToScrape, {
-        message: 'Importer error', error, status: AccountStatus.ERROR
-      }));
-      throw error;
-    }
-  });
-  await Promise.all(scrapingPromises);
+  const scrapePromises = scrapingConfig.accountsToScrape
+    .filter((accountToScrape) => accountToScrape.active !== false)
+    .map(async (accountToScrape) => ({
+      id: accountToScrape.id,
+      transactions: await fetchTransactions(accountToScrape, startDate, scrapingConfig.showBrowser, eventPublisher, dowloadedChrome)
+    }));
+
+  const promiseResults = await Promise.allSettled(scrapePromises);
```

Comment:
The main change is to use `Promise.allSettled(scrapePromises)`, for solving #171.

Instead of using a closure, I prefer to reduce the results from the promises. After trying to export the `map` content into a different function, I found the function is very similar to the `fetchTransactions`, just adds ErrorHandling and emits, so I merged those two functions.

Reply:
So now it considered Green even if some importers failed. WDYT?

![Screenshot from 2021-02-02 22-05-39](https://user-images.githubusercontent.com/17686879/106660157-52067c80-65a8-11eb-856e-b6952e0d9bcd.png)